src/main/java/com/es/phoneshop/model/product/ArrayListProductDao.java

```diff
@@ -23,22 +38,32 @@ public Product getProduct(Long id) throws ProductNotFoundException, IllegalArgum
     }
 
     @Override
-    synchronized public List<Product> findProducts() {
-        return products.stream()
-                .filter(isValidProduct())
-                .collect(Collectors.toList());
+    synchronized public List<Product> findProducts(String query, String sortBy, String order) {
+        List<Product> actualProducts;
+
+        if (query == null || query.trim().length() == 0) {
+            actualProducts = findAllProducts();
+        } else {
+            actualProducts = findProductsByQuery(query);
+        }
+
+        if (sortBy == null || order == null) {
```

Comment:
order may have a default value

Reply:
If order isn't important, parameter from get request will be null